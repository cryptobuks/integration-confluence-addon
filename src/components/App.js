import React, { Component } from 'react'
import Max from './Max'
import Min from './Min'
import Resize from './Resize'
import Link from './Link'
import ErrorMessage from './ErrorMessage'
import './app.css'

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      previewImgUrl: '',
      projectName: '',
      shareUrl: '',
      size: 'min',
      err: ''
    }
    this.toggleSize = this.toggleSize.bind(this)
  }

  componentDidMount() {
    const {apiUrl} = this.props.config
    const req = new Request(`${apiUrl}/api/share/${this.props.match.params.shareUrl}`)
    fetch(req)
      .then(response => {
        if (response.ok) {
          return response.json()
        }

        throw new Error(response.statusText)
      })
      .then(data => {
        const {previewImgUrl, projectName, shareUrl} = data
        this.setState({
          previewImgUrl,
          projectName,
          shareUrl
        })
      })
      .catch(err => {
        this.setState({
          err
        })
      })
  }

  toggleSize(evt) {
    const newSize = this.state.size === 'max' ? 'min' : 'max'
    this.setState({
      'size': newSize
    })
  }

  render() {
    if (this.state.err) {
      return (
        <div className="app ac-content">
          <ErrorMessage url={this.props.match.params.shareUrl} />
        </div>
      )
    }

    const Component = this.state.size === 'max' ? Max : Min
    return (
      <div className={`app ac-content ${this.state.size}`}>
        <Component
          previewImgUrl={this.state.previewImgUrl}
          projectName={this.state.projectName}
          shareUrl={this.state.shareUrl} />
        <div className="tools">
          <Resize size={this.state.size} action={this.toggleSize} />
          <Link shareUrl={this.state.shareUrl} />
        </div>
      </div>
    )
  }
}

export default App
