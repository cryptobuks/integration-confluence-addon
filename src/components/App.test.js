import React from 'react'
import renderer from 'react-test-renderer';
import App from './App'

it('renders correctly', () => {
  const InVisionConfig = {
    baseUrl: '__CONFLUENCE_BASE_URL__',
    apiUrl: '__CONFLUENCE_API_URL__'
  }
  const match = {
    params: {
      shareUrl: 'https://in.invisionapp.com/share/ATBWCTHR5'
    }
  }
  fetch.mockResponse(JSON.stringify({
    previewImgUrl: 'https://s3.invisionapp-cdn.com/storage.invisionapp.com/screens/thumbnails/235857145.png?x-amz-meta-iv=2&response-cache-control=max-age%3D2419200&x-amz-meta-mdi=0&x-amz-meta-ck=7b6ac614880f2a6dcb831703094006aa&AWSAccessKeyId=AKIAJFUMDU3L6GTLUDYA&Expires=1506816000&Signature=xSZfe1UhbXL%2FbsWq%2F5jn%2FuhT0Fc%3D',
    projectName: 'Paper embed prototypes',
    shareUrl: 'https://in.invisionapp.com/share/ATBWCTHR5'
  }))
  const tree = renderer.create(
    <App config={InVisionConfig} match={match} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
  const req = new Request(`${InVisionConfig.apiUrl}/api/share/${match.params.shareUrl}`)
  expect(fetch).toHaveBeenCalledWith(req)
})
