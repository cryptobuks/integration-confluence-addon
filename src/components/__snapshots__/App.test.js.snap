// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app ac-content min"
>
  <div
    className="wrapper"
  >
    <a
      href=""
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt=""
        className="thumb"
        onLoad={[Function]}
        src=""
      />
    </a>
    <div
      className="info"
    >
      <h4
        className="info--project-name"
      >
        
      </h4>
      <span
        className="info--project-type"
      >
        InVision Prototype
      </span>
    </div>
  </div>
  <div
    className="tools"
  >
    <a
      className="resize min"
      onClick={[Function]}
    >
      <span
        className="size"
      >
        <img
          alt="Resize"
          height="20"
          src="resize-full-screen.svg"
          width="20"
        />
      </span>
    </a>
    <a
      className="link"
      href=""
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="logo"
      >
        <img
          alt="InVisionApp logo"
          height="28"
          src="InVision-icon.svg"
          width="28"
        />
        <span
          className="open-in-invision"
        >
          Open in InVision
        </span>
      </span>
    </a>
  </div>
</div>
`;
